src/Sinch/Conversation/Messages/Message/ListMessage.cs

```diff
@@ -72,7 +74,7 @@ public sealed class ListSection
         /// <summary>
         ///     Gets or Sets Items
         /// </summary>
-        public List<IListItem> Items { get; set; }
+        public List<ListItem> Items { get; set; }
```

Comment:
Waiting feedback from dedicated team but seems all items should be of same type.
So this could imply to have something like `List<ListItem<T>>` where `T` could be `ListItemChoice` or `ListItemProduct`
(or even at `ListSection` level)

Reply:
Just received confirmation: All sections have to be of the same type in a ListMessage

Reply:
But, unfortunately, I don't see how can I simplify the structure anyway due to specifics of JSON structure of that response. 😕  So the way the things are middle-ground between specification and code, which I would like to merge as is anyway.

Reply:
What about using same design as the one used by SMS domain ? 
It could offer benefits:
1. using native OO C# language
2. same design across domains within the SDK (currently SMS is using  base class + inheritance for oneOf/allOf implementation)
3. once OO classes, this will enable to have ListItem<T> and ListSection<T> to restrict whole list items to be of same type

Out of this dedicated PR but highlighted by this one: point 1. and 2. are about whole Conversation domain. So design from one domain to another is not offering a consistant experience to SDK users